agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpServerTracer.java

```diff
@@ -195,6 +201,44 @@ protected void onRequest(final Span span, final REQUEST request) {
     // TODO set resource name from URL.
   }
 
+  protected void onConnectionAndRequest(Span span, CONNECTION connection, REQUEST request) {
+    String flavor = flavor(connection, request);
+    if (flavor != null) {
+      SemanticAttributes.HTTP_FLAVOR.set(span, flavor);
+    }
+    SemanticAttributes.HTTP_CLIENT_IP.set(span, clientIP(connection, request));
+  }
+
+  protected String clientIP(CONNECTION connection, REQUEST request) {
+    // try Forwarded
+    String forwarded = requestHeader(request, "Forwarded");
+    if (forwarded != null) {
+      // may be split by ; or ,
+      for (String forwardedValues : forwarded.split(";|(, *)")) {
```

Comment:
`split(regex)` is really slow, if we need a regex we should use `Pattern.compile` at least. But it'd be best if we can avoid split too due to the array allocations, do you think we can loop through to find the indexes instead of splitting?

@trask By the way, is it ok to use Guava here? `CharMatcher` can help with such searches.

Reply:
> is it ok to use Guava here?

Currently we don't, maybe better not to add this to our instrumentation api contract. We could expose something in our instrumentation api that wraps specific guava functionality if we need. I'd probably lean towards doing the index looping here for now (assuming we need to support the "Forwarded" header).

Reply:
Ah forgot this is the instrumentation API, also used in library instrumentation. Agree not to add dependencies.

Reply:
I'll change it to pattern.compile and incorporate the trim optimisation